faunadb-scala/src/test/scala/faunadb/ClientSpec.scala

```diff
@@ -920,7 +920,131 @@ class ClientSpec
       // Identify
       val identifyR = client.query(Identify(createR(RefField), "abcdefg")).futureValue
       identifyR.to[Boolean].get shouldBe true
-      }
+   }
+
+  it should "create_access_provider successful creation" in {
+    val roleName = "my_role" + aRandomString
+    val providerName = "my_provider" + aRandomString
+    val issuerName = "my_issuer" + aRandomString
+    val fullUri = s"https://${aRandomString}.auth0.com"
+
+    val roleV = adminClient.query(CreateRole(Obj(
+      "name" -> roleName,
+      "privileges" -> Obj(
+        "resource" -> Databases(),
+        "actions" -> Obj("read" -> true)
+      )
+    ))).futureValue
+
+    val providerV = adminClient.query(CreateAccessProvider(Obj(
+      "name" -> providerName,
+      "issuer" -> issuerName,
+      "jwks_uri" -> fullUri,
+      "roles" -> ArrayV(roleV(RefField))))
+    ).futureValue
+
+    providerV("roles").to[Seq[RefV]].get.size shouldBe 1
```

Comment:
And, we could also leverage ScalaTest's ["aggregations" matchers](https://www.scalatest.org/user_guide/using_matchers#workingWithAggregations) here to verify the `membership`/`roles` field more accurately: 

```scala
providerV("membership").to[Seq[RefV]].get should contain theSameElementsAs(Seq(Role(roleName)))
```

:)

Reply:
Neat :+1: 

I went with
```
providerV("roles").to[Seq[RefV]].get should contain theSameElementsAs(Vector(roleV(RefField).get))
```

Reply:
Oh, perfect, yeah, we need to assert everything against values! :)